index.bs

```diff
@@ -368,7 +364,32 @@ The <dfn method for="XRSession">end()</dfn> method provides a way to manually sh
 
 </div>
 
-<div class="algorithm" data-algorithm="request-frame-of-reference">
+Each {{XRSession}} has a <dfn>list of pending render states</dfn>, which is initially empty.
+
+<div class="algorithm" data-algorithm="update-render-state">
+
+When the <dfn method for="XRSession">updateRenderState(|newState|)</dfn> method is invoked, the user agent MUST return [=a new Promise=] |promise| and run the following steps [=in parallel=]:
+
+  1. Let |session| be the target {{XRSession}}.
+  1. If |session|'s [=list of animation frame callbacks=] is empty, [=update the render state=] with |newState| and [=/resolve=] |promise|.
+  1. Else append |newState| to |session|'s [=list of pending render states=], associated with |promise|.
+
+</div>
+
+<div class="algorithm" data-algorithm="apply-pending-render-states">
+
+When an {{XRSession}} has finished processing a batch of [=XR animation frames=], it MUST <dfn>apply pending render states</dfn> by running the following steps:
+
+  1. Let |session| be the target {{XRSession}}.
+  1. Let |renderStates| be a list of the entries in |session|'s [=list of pending render states=], in the order in which they were added to the list.
+  1. Set |session|'s [=list of pending render states=] to the empty list.
+  1. For each entry |newState| in |renderStates|, in order:
+    1. [=Update the render state=] with |newState|
```

Comment:
On my first read-through I got confused between `[=Update the render state=]` which is defined as `update-renderstate` and the section above defining the behavior of `updateRenderState()` which is defined as `update-render-state`.  Can we rename one of them?  Perhaps the former should be `[=Apply individual render state=]` with the algorithm name of `apply-individual-render-state`?  Or something?

Reply:
Yeah, the naming was more confusing than I realized. Thanks for pointing it out, new commit cleans that up a lot by reducing the number of separate algorithms used.